frame/elections-phragmen/src/lib.rs

```diff
@@ -524,10 +503,13 @@ decl_event!(
 		Balance = BalanceOf<T>,
 		<T as frame_system::Trait>::AccountId,
 	{
-		/// A new term with new members. This indicates that enough candidates existed, not that
-		/// enough have has been elected. The inner value must be examined for this purpose.
+		/// A new term with new members. This indicates that enough candidates existed to run the
+		/// election, not that enough have has been elected. The inner value must be examined for
+		/// this purpose. A `NewTerm([])` indicates that some candidates got their bond slashed and
```

Comment:
Why isn't that an `NewTerm(None)` to indicate the different behavior? Or even another event.

Reply:
I don't see any difference here between `[]` and `None`. I think it only adds few bits of encoding overhead.

Reply:
What we could do is to get rid of EmptyTerm and replace 
- `EmptyTerm` -> `NewTerm(None)`: we had no members and no candidates and did nothing. 
- `NewTerm(x)` where x is of any length -> `NewTerm(Some(x))`: we had some members and did some stuff but maybe none were elected. 

This _some stuff_ is very important. If we have no candidates to begin with (`EmptyTerm`) we keep the previous members. 
If we have some candidates but they fail, this is a _new term_ and we remove the previous members, call into `T::ChangeMembers` potentially with an empty new set. 

These two are very different.

Reply:
> I don't see any difference here between `[]` and `None`. I think it only adds few bits of encoding overhead.

I think it expresses the difference much better in the code, but it was just some suggestion ;)